worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/DeviceMonitor.scala

```diff
@@ -21,7 +21,7 @@ import java.io._
 import java.nio.charset.Charset
 import java.util
 import java.util.{Set => jSet}
-import java.util.concurrent.{ConcurrentHashMap, TimeUnit}
+import java.util.concurrent.{ConcurrentHashMap, CopyOnWriteArrayList, TimeUnit}
```

Comment:
I think CopyOnWriteArrayList is inefficient, we can just use ConcurrentSet, it's rare that multiple noncritical errors are reported at the same millisecond. If that happens, I think it's fine to count them as one.